infrastructure/cdn-in-a-box/traffic_ops/to-access.sh

```diff
@@ -168,85 +172,68 @@ to-enroll() {
 	else
 		export MY_CDN="CDN-in-a-Box"
 	fi
+	if [[ ! -z "$3" ]]; then
+		export MY_CACHE_GROUP="$3"
+	else
+		export MY_CACHE_GROUP="CDN_in_a_Box_Edge"
+	fi
+	if [[ ! -z "$4" ]]; then
+		export MY_TCP_PORT="$4"
+	else
+		export MY_TCP_PORT="80"
+	fi
+	if [[ ! -z "$5" ]]; then
+		export MY_HTTP_PORT="$5"
+	else
+		export MY_HTTP_PORT="443"
+	fi
 
 	export MY_NET_INTERFACE='eth0'
 	export MY_DOMAINNAME="$(dnsdomainname)"
-	export MY_IP="$(ifconfig $MY_NET_INTERFACE | grep 'inet ' | tr -s ' ' | cut -d ' ' -f 3)"
+	MY_IP="$(ifconfig $MY_NET_INTERFACE | grep 'inet ' | tr -s ' ' | cut -d ' ' -f 3)"
+	export MY_IP=${MY_IP#"addr:"}
 	export MY_GATEWAY="$(route -n | grep $MY_NET_INTERFACE | grep -E '^0\.0\.0\.0' | tr -s ' ' | cut -d ' ' -f2)"
-	export MY_NETMASK="$(ifconfig $MY_NET_INTERFACE | grep 'inet ' | tr -s ' ' | cut -d ' ' -f 5)"
-	export MY_IP6_ADDRESS="$(ifconfig $MY_NET_INTERFACE | grep inet6 | grep global | awk '{ print $2 }')"
+	MY_NETMASK="$(ifconfig $MY_NET_INTERFACE | grep 'inet ' | tr -s ' ' | cut -d ' ' -f 5)"
+	export MY_NETMASK=${MY_NETMASK#"Mask:"}
+	MY_IP6_ADDRESS="$(ifconfig $MY_NET_INTERFACE | grep inet6 | grep -i global | sed 's/addr://' | awk '{ print $2 }')"
+	export MY_IP6_ADDRESS=${MY_IP6_ADDRESS%%/*}
 	export MY_IP6_GATEWAY="$(route -n6 | grep UG | awk '{print $2}')"
 
 	case "$serverType" in
 		"edge" )
 			export MY_TYPE="EDGE"
 			export MY_PROFILE="ATS_EDGE_TIER_CACHE"
 			export MY_STATUS="REPORTED"
-			if [[ ! -z "$3" ]]; then
-				export MY_CACHE_GROUP="$3"
-			else
-				export MY_CACHE_GROUP="CDN_in_a_Box_Edge"
-			fi
 			;;
 		"mid" )
 			export MY_TYPE="MID"
 			export MY_PROFILE="ATS_MID_TIER_CACHE"
 			export MY_STATUS="REPORTED"
-			if [[ ! -z "$3" ]]; then
-				export MY_CACHE_GROUP="$3"
```

Comment:
I think this will break. The mid's `run.sh` is calling `to-enroll` without specifying a Cache Group. Haven't looked at any of the others, but I suspect they do the same thing - which in their case is fine because it by and large won't matter which Cache Group they are placed into.

Reply:
I don't quite understand what you mean
The `grep to-enroll` results are shown as follows:

```
edge/run.sh:61:to-enroll edge $CDN || (while true; do echo "enroll failed."; sleep 3 ; done)
mid/run.sh:61:to-enroll mid $CDN || (while true; do echo "enroll failed."; sleep 3 ; done)
traffic_monitor/run.sh:63:to-enroll tm $CDN || (while true; do echo "enroll failed."; sleep 3 ; done)
traffic_ops/run.sh:130:to-enroll "to" ALL || (while true; do echo "enroll failed."; sleep 3 ; done)
traffic_ops/run-go.sh:79:to-enroll "to" ALL || (while true; do echo "enroll failed."; sleep 3 ; done)
traffic_portal/run.sh:70:to-enroll "tp" ALL || (while true; do echo "enroll failed."; sleep 3 ; done)
traffic_router/run.sh:100:to-enroll tr || (while true; do echo "enroll failed."; sleep 3 ; done)
```

Except TR, all another services have specified their cache group.
Take mid as an example, it has specified its cache group as `CDN-in-a-Box`.
Please refer to [here](https://github.com/apache/trafficcontrol/blob/master/infrastructure/cdn-in-a-box/mid/run.sh#L48-L61) for details: 

On the other hand, what is the function of cache group?
Could you share the document link with me?

Reply:
no, `$CDN` and `ALL` are the names of CDNs, not Cache Groups. Those are all relying on the defaults

Reply:
@ocket8888 I see what you mean. I'm sorry to have kept you waiting so long.

As the above search results, there is no any service supply more than 2 parameters.
It means current design wants to put them all in the same cache group, even that's not a good idea.

This commit add an *empty string* for the third prameter when TV calls to-enroll, in [here](https://github.com/apache/trafficcontrol/pull/3287/commits/2df03f70eba694a671b0046648d621e0da98566f#diff-5e6d2212372d15ad9c4e202c58843b85R29)
So this commit will not break anything because it keep the original architecture and give default value to each new parameter.

How about let me raise a new issue for it first?
Maybe I can fix it after I know more about the cache group.

Reply:
You can read more about Cache Groups [in their section of our official documentation](https://traffic-control-cdn.readthedocs.io/en/latest/faq/administration.html#cachegroup) (though that'll move soon if I get my PR merged).

In the meantime, lemme take another look at this PR.

Reply:
Yeah, the problem still exists. You can see the issue in `infrastructure/cdn-in-a-box/mid/run.sh` on line #61 where it's calling `to-enroll`:

```bash
to-enroll mid $CDN || (while true; do echo "enroll failed."; sleep 3 ; done)
```

This means that when `to-enroll` is called, it sees its positional parameters defined as:

```bash
1="mid"
2="$CDN"
3=""
4=""
5=""
# etc.
```

so with your change the Cache Group to which the mid will be assigned will be set to `"CDN_in_a_Box_Edge"`, which is not correct. The Mid-tier cache server needs to be in a separate Cache Group from the Edge-tier cache server, so that the the latter can assign the former as its parent.

Reply:
OMG I found what's my problem! It's all my fault!
I did not see clearly that only mid cache group is set to "CDN_in_a_Box_**Mid**" and all the other are set to "CDN_in_a_Box_**Edge**"!
I'll fix it right away!